### PR TITLE
`Development`: Remove unused repository methods

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

We don't need the methods, so we can remove them. Also added an architecture test to prevent unused methods in repositories.

### Description

I ran an inspection on the project and auto-removed all occurrences of unused methods.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by nityanandaz — APPROVED** [review]
Thanks for cleaning

**Review by egekurt123 — APPROVED** [review]
LGTM

File: src/test/java/de/tum/in/www1/artemis/architecture/ArchitectureTest.java [hunk 1]
```diff
@@ -123,6 +124,22 @@ void testCorrectServiceAnnotation() {
 
     }
 
+    @Test
+    void testNoUnusedRepositoryMethods() {
+        ArchRule unusedMethods = noMethods().that().areAnnotatedWith(Query.class).and().areDeclaredInClassesThat().areInterfaces().and().areDeclaredInClassesThat()
+                .areAnnotatedWith(Repository.class).should(new ArchCondition<>("not be referenced") {
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/architecture/ArchitectureTest.java:130]:
```suggestion
                .areAnnotatedWith(Repository.class).should(new ArchCondition<>("be referenced") {
```

**Review by Strohgelaender — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]
Maintainer approved

**Pull request merged**